sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/SapServiceBindingsProperties.java

```diff
@@ -0,0 +1,132 @@
+package com.sap.cloud.environment.servicebinding;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+/**
+ * Properties of service bindings defined in a properties file such as Spring's default file 'application.properties'.
+ */
+public class SapServiceBindingsProperties {
+    /**
+     * default plan name
+     */
+    private static final String BINDING_PLAN = "standard";
+
+    /**
+     * properties of configured service bindings
+     */
+    private Map<String, ServiceBindingProperties> serviceBindings;
+
+    public Map<String, ServiceBindingProperties> getServiceBindings() {
+        return serviceBindings;
+    }
+
+    public void setServiceBindings(Map<String, ServiceBindingProperties> serviceBindings) {
+        this.serviceBindings = serviceBindings;
+    }
+
+    public static class ServiceBindingProperties {
+        /**
+         * properties for credentials
+         */
+        private Map<String, Object> credentials;
+
+        /**
+         * label of the service binding
+         */
+        private String label;
+
+        /**
+         * tags of the service binding
+         */
+        private String[] tags;
+
+        /**
+         * name of the service binding
+         */
+        private String name;
+
+        /**
+         * name of the service
+         */
+        private String serviceName;
+
+        /**
+         * name of the instance
+         */
+        private String instanceName;
+
+        /**
+         * plan used to create the service binding
+         */
+        private String plan = BINDING_PLAN;
+
+        public Map<String, Object> getCredentials() {
+            if (credentials == null) {
+                return Collections.emptyMap();
+            } else {
+                /* the key names are converted to lower-case */
+                return credentials
+                        .entrySet()
+                        .stream()
+                        .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().toLowerCase(), entry.getValue()))
+                        .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
```

Comment:
Why do you normalize credentials? This might not fit with the expectation that developers have about their credential properties in the service binding. These are usually case-sensitive for all other cases (VCAP_SERVICES or Kyma bindings)

Reply:
that's a good question!
I noticed that all property names at the bindings are lower-case. And the used libraries search for the bindings' properties in lower-case.
But Spring application properties names are usually in lower-camel-case.
I fell into this trap when externalizing these VCAP properties and adding them to my test `application.yaml` file. Then the bindings were not detected and/or the credentials could not be read.
I assumed a standard Spring Boot developer will do the same mistake. That's why I normalized the property keys.
Of course not the values...

Reply:
> And the used libraries search for the bindings' properties in lower-case.

Can you point to that? Do they just happen to use lower-case property names or are they explicitly lowercasing their property keys before looking into the binding properties?

I am wondering if that is really the case for all libraries. In my opinion it should be possible to get the same property keys that you would get via VCAP_SERVICES, and a service might decide to use uppercase or camelcase property names. I wouldn't enforce any normalization, but keep the responsibility to the developer to add the property keys in the right and expected format.

Reply:
I look it up (that will take some time)

in general I completely agree to your opinion (and at places where these properties are looked up, it should be normalized to accept different 'styles' of property names)- but that was not the case in my tests...)

Reply:
Okay, but then that's a "bug"/limitation in these other libraries. A very generic service binding library like this shouldn't make such assumptions in my opinion.

Reply:
OK, let's remove it and I search for the places, where I think those libs should be improved.